maven-core/src/main/java/org/apache/maven/lifecycle/mapping/LifecycleMappingImpl.java

```diff
@@ -0,0 +1,62 @@
+package org.apache.maven.lifecycle.mapping;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Alternate implementation of {@link LifecycleMapping} as default one is too much plexus-suited.
+ */
+public final class LifecycleMappingImpl
+    implements LifecycleMapping
+{
+  private final Map<String, Lifecycle> lifecycles;
+
+  public LifecycleMappingImpl( final List<Lifecycle> lifecycles )
+  {
+    this.lifecycles = Collections.unmodifiableMap(
+        lifecycles.stream().collect( toMap( Lifecycle::getId, l -> l ) )
+    );
+  }
+
+  @Override
+  public Map<String, Lifecycle> getLifecycles()
+  {
+    return lifecycles;
+  }
+
+  @Deprecated
```

Comment:
as usual why is deprecated ...

maybe if it must stay in interface use default  methods .. so child class will needn't implement it

Reply:
Iface has them deprecated, but DefaultLifecycleMapping does something funky (and has deprecated `phases` member as well) so I left it. 

BUT, if you look at Plexus XML, that field is NEVER injected (as here it was used the fact, a "niche" capability of Plexus, to inject private fields). So to say, as I see, `phases` despite having some logic around, the field itself was never injected (in Java code is impossible, unless using reflection, and Plexus XML never had config for that field).

Reply:
ok, its look like another refactor

Reply:
This class is gone